src/poetry/core/semver/version_union.py

```diff
@@ -238,6 +238,62 @@ def _ranges_for(
 
         raise ValueError(f"Unknown VersionConstraint type {constraint}")
 
+    def _exclude_single_wildcard_range_string(self) -> str:
+        """
+        Helper method to convert
+        """
+        assert self.excludes_single_wildcard_range()
```

Comment:
We should raise an exception here. Relying on an `assert` seems to be a no-go to me, since asserts are not executed when running python in optimized mode.